### ISSUE TITLE
Is there a keyboard shortcut or setting for cycling through workspaces?

### ISSUE THREAD

**Issue by s7dhansh** [linuxdeepin/developer-center#1035]
Either, a setting that can cause Win+Right to go to first workspace if currently on the last one.
Or a command to cycle through workspaces which can be assigned a custom key?

I was unable to find anything in the control center or google.

**Comment by antechdesigns** [comment]
should be superkey+left or right arrow?

**Comment by listenerri** [comment]
Please wait next release edition which will use kwin as window manager, may be is 15.9.3 : )

**Comment by s7dhansh** [comment]
> @s7dhansh
> should be superkey+left or right arrow?

Yes but they do not **cycle through**.

**Comment by keybreak** [comment]
Oh...I actually like it this way though, but if we could have some option or special shorcut for this - will be very cool! :smiley:

**Comment by aiamuzz** [comment]
FYI ... one could use 'WinKey+1,2,3,4,5(workspace number)' will directly take you to the particular workspace ... try this out for a few days and you'll forget the need for a cyclic navigation or 'Winkey + right/left' option all together !!!

**Comment by aiamuzz** [comment]
further ...

To move a particular application window to a particular workspace ... just use

Winkey+Shift+Workspace No. you want to send the application window to.

Eg. 'Winkey+Shift+3' will send the active application window from the current workspace to workspace 3 !!!

pretty handy shortcuts to make the most out of windows and workspaces ...

**Comment by s7dhansh** [comment]
@aiamuzz that may be true when you have many workspaces and you manually manage them. it is futile to assume that something that works for you will work for everyone. For instance, what works for me is having a minimalistic
1. 2 workspace system (similar to 2 monitor system - great for development), on xfce
2. devilspie to ensure browser and related windows on one ws while editor etc on another
3. a Win+` for cycling (and an Alt + Win + ` for cycling windows - though it is mostly redundant with devilspie)
4. an omnipresent tilix quake terminal

3 is a great piece that ensures i do not have to remember which workspace i am on or what key i have to hit to switch. It is also the reason why i went back to xfce after trying and liking deepin. You too may try this system and may be you will *forget* the need of having so many shortcuts, or so many workspaces altogether :wink:.

**Comment by aiamuzz** [comment]
@s7dhansh ...

well ... works for me, works for you, ... is a slippery slope ... the eternal truth is that this can never be conquered to everyone's satisfaction !!!

we just share things here to see if something will help someone else ... 

FYI ... earlier i used to use 'Winkey+right/left' and found it good enough ... and then i accidentally tripped over 'Winkey+worskpace number' and 'Winkey+shift+workspace number' ... i found this far more efficient than the previous system i thought was already good ... **so you see i have kept myself OPEN to newer things and tried to adapt and subsequently found efficiency in what was already available ... so you see nothing is MY way ... its what was available and how i made the most out of it** ... adapting to the existing system should be the first act for any user on a newer system ... trying to look for our old system in a new system often leads to dissatisfaction !!!

some of my personal experiences on how i adapted to a newer system ...

When I was introduced to Deepin at first i didn't have any previous experience of working with WS's though i was on UbuntuGnome(which to had workspaces ) ...

I then discovered that deepin had implemented rather great system of managing WS based ...

1. they had one shortcut to view all the open application windows in a given workspaces
2. another shortcut to view all open application windows across all workspaces
3. one more shortcut to view all workspaces and way to move application between them(drag and drop)

i was instantly drawn to this system the above 3 options were great, but one could not access them through the keyboard, they were only accessible through the mouse(click, drag and drop) ... i raised a feature request and now one can use the 'right/left arrow keys' or he 'tab key' to navigate and choose the application they want to be active.

> 3\. a `Win+tilde` for cycling (and an `Alt + Win + tilde` for cycling windows - though it is mostly redundant with devilspie)

so you see we can achieve similar navigation you speak of in your point 3 on deepin already ... test drive it for sometime ... we can only share our experiences ... hoping the best ...

**Comment by aiamuzz** [comment]
> Please wait next release edition which will use kwin as window manager, may be is 15.9.3 : )

@listenerri ... oh really ... is it a radical change, a complete overhaul of the current window manager ? 
BTW what is the current WM on deepin based on ... is it based on an existing project or is it a deepin's own creation ?

any screenshots of this kwin WM ... that you can share ?

**Comment by keybreak** [comment]
@aiamuzz 
It's faster and should solve a lot of problems / visual bugs we have now.

**Comment by aiamuzz** [comment]
> @aiamuzz
> It's faster and should solve a lot of problems / visual bugs we having now.

oh ... i was watching some videos ... i hope its NOT implemented as it is(default) ...

If i am not wrong ... UbuntuGnome too had mutter just like we have it on Deepin ... but the way @shule1987  ... had made it elegant ... i just couldn't believe it was the same WM i had on gnome ... from not using the WM in UbuntuGnome to a big fan of it on Deepin is all credits to @shule1987 and his touch of elgance.

I really really hope @shule1987  ... has worked his elegance on kwin WM too ... if not its better you don't launch until he has ... the current system is elegant enough !!!

**I would really love to see @shule1987 ... take on making kwin elegant enough to fit into the overall elegance of Deepin OS !!!**

**Comment by keybreak** [comment]
Don't worry, it's should be more technical aspects, not aesthetics.
I'd also expect aesthetics & design not to change drastically, as it is really lovely now! :smile_cat:

**Comment by justforlxz** [comment]
Sorry, this issue will be closed soon. If it is necessary to discuss it again, please create a new issue.
